src/PreserveModuleNamePlugin.ts

```diff
@@ -72,90 +164,320 @@ function getPreservedModules(modules: Webpack.Module[]) {
+    const resourceKeys: string[] = Object.keys(moduleResources);
+    if (!moduleEntry && resourceKeys.length === 1) {
+      moduleEntry = moduleResources[resourceKeys[0]].relative;
+    }
+
+    // Map the resources to the module id
+    resourceKeys.forEach((resource: string) => {
+      const data: NodeModule.Data = moduleResources[resource];
+
+      // Entry?
+      if (moduleEntry === data.relative) {
+        _nodeModuleResourceIdMap[resource] = moduleKey;
+        return;
+      }
+
+      // Build the id from the resources common path
+      let key: string = data.relative.replace(new RegExp(`^${escapeString(commonPath)}`), '');
+      key = (key.startsWith('/')) ? key : `/${key}`;
+      _nodeModuleResourceIdMap[resource] = `${moduleKey}${key}`;
+    });
+  }
+}
+
+/**
+ *  Find the path to a modules resource relative to the webpack
+ *  resolve module paths
+ *
+ *  @param  {Webpack.Module} module The module to get the relative path
+ *  @param  {string[]} paths The webpack resolver module paths
+ *
+ *  @return {string|null} The relative path if available, null otherwise
+ */
+function getRelativeModule(module: Webpack.Module, paths: string[]): string | null {
+  if (!module || !module.resource || !paths || !paths.length) {
+    return null;
+  }
+
+  // Try to find the module in the resolver paths
+  for (let i = 0, len = paths.length; i < len; i++) {
+    const relative: string = path.relative(paths[i], module.resource);
+    if (!relative.startsWith('..')) {
+      return relative;
+    }
+  }
+
+  // Nothing relative
+  return null;
+}
+
+/**
+ *  Find the path to a modules resource relative to webpack aliases
+ *
+ *  @param  {Webpack.Module} module The module to alias
+ *  @param  {[{ [key: string]: string }|null} aliases The webpack aliases
+ *
+ *  @return {string|null} The alias path if available, null otherwise
+ */
+function getAliasModule(module: Webpack.Module, aliases: { [key: string]: string } | null): string | null {
+  if (!module || !module.resource || !aliases || !Object.keys(aliases).length) {
+    return null;
+  }
+
+  // Look for the module in each alias
+  for (let alias in aliases) {
+    const relative: string = path.relative(path.resolve(aliases[alias]), module.resource);
+    if (relative.startsWith('..')) {
+      continue;
+    }
+
+    // Absolute alias?
+    alias = alias.replace(/\$$/, '');
+    return (relative && relative.length) ? `${alias}/${relative}` : alias;
+  }
+
+  // Nothing aliased
+  return null;
+}
+
+/**
+ *  Get the module id based on its resource
+ *
+ *  @param  {Webpack.Module} module The module to get the id for
+ *  @param  {string[]} paths The webpack resolver module paths
+ *  @param  {[{ [key: string]: string }|null} aliases The webpack aliases
+ *
+ *  @return {string|null} The module id if available, null otherwise
+ */
+function getModuleId(module: Webpack.Module, paths: string[], aliases: { [key: string]: string } | null): string | null {
```

Comment:
When extracting this function, we lost one bit of functionality.
If `module[preserveModuleNode]` was defined it would be used without looking at anything else.
This can be set by some plugins when creating the dependency, if I recall correctly.

Is this intentionnal?

Reply:
I was unaware that other modules could set it. The `AureliaDependenciesPlugin` uses it to explicitly set the generated moduleId.